### PR TITLE
GUI-2640 - Upgrade to AngularJS 1.5.x

### PR DESCRIPTION
**Pull request by coyote240**
https://eucalyptus.atlassian.net/browse/GUI-2640

### CODE REVIEW

**Comment by aliceh** [comment]
Retest this please

**Comment by aliceh** [comment]
Retest this please

**Comment by aliceh** [comment]
retest this please

**Comment by aliceh** [comment]
Retest this please

**Comment by aliceh** [comment]
Retest this please

File: eucaconsole/forms/instances.py [hunk 1]
```diff
@@ -181,7 +181,7 @@ def set_choices(self, request):
         self.keypair.choices = self.get_keypair_choices()
         self.securitygroup.choices = self.choices_manager.security_groups(
             securitygroups=self.securitygroups, use_id=True, add_blank=False)
-        self.role.choices = ChoicesManager(self.iam_conn).roles(add_blank=True)
+        self.role.choices = ChoicesManager(self.iam_conn).roles(add_blank=False)
```
Comment (dkavanagh) [inline eucaconsole/forms/instances.py]: role is not required, so is the widget allowing no selection now?

File: eucaconsole/static/js/pages/cloudwatch_charts.js [hunk 2]
```diff
@@ -87,10 +87,9 @@ angular.module('CloudWatchCharts', ['EucaConsoleUtils', 'ChartAPIModule', 'Chart
 })
 .controller('CloudWatchChartsCtrl', function ($scope, eucaUnescapeJson, eucaOptionsArray, ModalService) {
     var vm = this;
-    vm.duration = 3600;  // Default duration value is one hour
+    vm.duration = '3600';  // Default duration value is one hour
```
Comment (dkavanagh) [inline eucaconsole/static/js/pages/cloudwatch_charts.js]: Is there a reason this is made into a string, but ones below are not?

File: eucaconsole/static/js/pages/instance_launch.js [hunk 3]
```diff
@@ -39,7 +39,7 @@ angular.module('LaunchInstance', ['TagEditor', 'BlockDeviceMappingEditor', 'Imag
         $scope.selectedGroupRules = {};
         $scope.securityGroupModal = $('#create-securitygroup-modal');
         $scope.securityGroupForm = $('#create-securitygroup-form');
-        $scope.securityGroupChoices = {};
+        $scope.securityGroupChoices = [];
```
Comment (dkavanagh) [inline eucaconsole/static/js/pages/instance_launch.js]: Are the changes here something I forgot to back out?

File: eucaconsole/static/js/thirdparty/foundation-datepicker/foundation-datepicker.js [hunk 4]
```diff
@@ -1273,9 +1301,9 @@
         },
         headTemplate: '<thead>' +
             '<tr>' +
-            '<th class="prev"><i class="fa fa-chevron-left"/></th>' +
+            '<th class="prev"><i class="fa fa-chevron-left fi-arrow-left"/></th>' +
```
Comment (dkavanagh) [inline eucaconsole/static/js/thirdparty/foundation-datepicker/foundation-datepicker.js]: I remember removing the fi- icon refs manually in the past. They somehow interfered with our fa- icons showing up.

File: eucaconsole/forms/instances.py (hunk 1 above)
Comment (dkavanagh) [inline eucaconsole/forms/instances.py]: I found the place where the blank option is added in the markup.

File: eucaconsole/templates/instances/instance_launch.pt [hunk 5]
```diff
@@ -156,7 +156,9 @@
                                     <label class="right" i18n:translate="">Role</label>
                                 </div>
                                 <div class="large-8 small-8 columns field">
-                                    <select id="role" ng-options="k as v for (k, v) in roleList" ng-model="role" name="role"></select>
+                                    <select id="role" ng-options="k as v for (k, v) in roleList track by v" ng-model="role" name="role">
+                                        <option value="" disabled="" selected="">Select...</option>
```
Comment (dkavanagh) [inline eucaconsole/templates/instances/instance_launch.pt]: should be marked for i18n

File: eucaconsole/forms/instances.py (hunk 1 above)
Comment (coyote240) [inline eucaconsole/forms/instances.py]: Yeah, was having a tough time making the widget and the select work together, so opted for the client-side only option.

File: eucaconsole/static/js/pages/cloudwatch_charts.js (hunk 2 above)
Comment (coyote240) [inline eucaconsole/static/js/pages/cloudwatch_charts.js]: This is one of the first issues I found.  In this case, the seconds are used as a string, and angular was more strict about it.  Still seeking to understand.

File: eucaconsole/static/js/pages/instance_launch.js (hunk 3 above)
Comment (coyote240) [inline eucaconsole/static/js/pages/instance_launch.js]: I did this.

File: eucaconsole/static/js/thirdparty/foundation-datepicker/foundation-datepicker.js (hunk 4 above)
Comment (coyote240) [inline eucaconsole/static/js/thirdparty/foundation-datepicker/foundation-datepicker.js]: This is actually a third-party file.  This change came with the updates.

Comment (dkavanagh) [inline eucaconsole/static/js/thirdparty/foundation-datepicker/foundation-datepicker.js]: yes, and I needed to edit the supplied file. You can test by looking for the "month shift" buttons on the date picker to see if they have the chevrons.

Comment (coyote240) [inline eucaconsole/static/js/thirdparty/foundation-datepicker/foundation-datepicker.js]: I feel like we're better off overriding the definition in our own CSS than altering third-party js files.

**Pull request merged**